### PR TITLE
Prioritize class method if there is duplicated attribute name

### PR DESCRIPTION
**Pull request by yanboliang**
Fix https://github.com/pytorch/torchdynamo/issues/276

### CODE REVIEW

File: torchdynamo/variables/nn_module.py [hunk 1]
```diff
@@ -105,13 +105,20 @@ def var_getattr(self, tx, name):
         if not self.source:
             unimplemented("GETATTR with no source")
 
-        if name in base_dict:
+        # class attribute has the highest priority
+        try:
+            subobj = inspect.getattr_static(base.__class__, name)
+            class_member = False
```
Comment (yanboliang) [inline torchdynamo/variables/nn_module.py:111]: I still use ```class_member``` but I think it's a confusing name, ```object_member``` should be better? Correct me if I'm wrong. @jansel

**Comment by jansel** [comment]
So normal python objects have the opposite behavior
```
>>> class Foo:
...   def __init__(self):
...     self.hook = lambda: "on the object"
...   def hook(self):
...     return "on the class"
... 
>>> Foo().hook()
'on the object'
>>> 
```


Trying a bit more with `nn.Modules`:
```
>>> class Bar(torch.nn.Module):
...   def __init__(self):
...     super().__init__()
...     self.hook1 = torch.nn.ReLU()
...     self.hook2 = lambda: "on the object"
...     self.hook3 = "on the object"
...   def hook1(self):
...     return "on the class"
...   def hook2(self):
...     return "on the class"
...   def hook3(self):
...     return "on the class"
... 
>>> Bar().hook1
<bound method Bar.hook1 of Bar(
  (hook1): ReLU()
)>
>>> Bar().hook2()
'on the object'
>>> Bar().hook3
'on the object'
>>> 
```

As far as I can tell the object one wins.  What is different in this case?

**Review by jansel — CHANGES_REQUESTED** [review]
We should update the logic to match what is in the linked comment above.

**Comment by yanboliang** [comment]
See more details at https://github.com/pytorch/torchdynamo/issues/276#issuecomment-1136550186

**Review by jansel — APPROVED** [review]

**Pull request merged**